### ISSUE TITLE
Forbid function calls in multiple lines

### ISSUE THREAD

**Issue by thalesmello** [pagarme/javascript-style-guide#12]
```js
// Bad
const responseBody = singleErrorResponse(
    req,
    'action_forbidden',
    null,
    `Transação com status '${transaction.status}' não pode ser estornada`
)

// Good
const type = 'action_forbidden'
const param = null
const message =  `Transação com status '${transaction.status}' não pode ser estornada`
const responseBody = singleErrorResponse(req, type, param, message)

// Good
const responseBody = singleErrorResponse({
    req,
    type: 'action_forbidden',
    parameter: null,
    message: `Transação com status '${transaction.status}' não pode ser estornada`
})

// Good
const even = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].filter(number => {
  return number % 2 === 0
})
```

**Comment by derekstavis** [comment]
I don't agree with not allowing to split function arguments.

This is necessary when you integrate with third party APIs that you don't have control to.

**Comment by thalesmello** [comment]
For those situations, you can do as in the example:
```
// Good
const type = 'action_forbidden'
const param = null
const message =  `Transação com status '${transaction.status}' não pode ser estornada`
const responseBody = singleErrorResponse(req, type, param, message)
```

**Comment by mccraveiro** [comment]
@thalesmello I also don't agree with this.
Have you seen our core routes definition? Each route has multiple middlewares that are passed as parameters. Breaking them into lines brings better readability. Also, your example above can't be reproduced there as middlewares are already constants.

**Comment by thalesmello** [comment]
Example? @mcraveiro

**Comment by mccraveiro** [comment]
@thalesmello I can't post here as it is from a private repo, but... something similar to this:
```js
app.get(
  '/users',
  middlewareA,
  middlewareB,
  middlewareC,
  middlewareD,
  usersController
)
```
vs
```js
app.get('/users', middlewareA, middlewareB, middlewareC, middlewareD, usersController)
```

PS: middlewares names are much longer as they have semantic meaning.

**Comment by thalesmello** [comment]
You could do:

```js
const loggedUser = [
  middlewareA,
  middlewareB,
  middlewareC,
  middlewareD,
]

app.get('/users', ...loggedUser, usersController)
```

Or even better, we could use the adequate express abstraction for this.

```js
const loggedUser = express.Router()
loggedUser.use(middlewareA)
loggedUser.use(middlewareB)
loggedUser.use(middlewareC)
loggedUser.use(middlewareD)

app.get('/users', loggedUser, usersController)
```

Repeating the same point I said in #10, splitting function call across
multiple lines leads to bad design. It's much more convenient to break
your function call across multiple lines rather than giving good names
for the abstractions in your systems. 

However, I will choose the second option anytime, as it leads to a code that's
easier to read and understand. Using names for variables is documentation,
which will make it easier for people to read and understand your code.

**Comment by thalesmello** [comment]
Another plus side, we would also get a consistent dangling comma behavior. Special cases aren't special enough to break the rules.

**Comment by thalesmello** [comment]
@derekstavis @MarcoWorms @grvcoelho May I get your comments?

**Comment by MarcoWorms** [comment]
Or even better

```js
const handleUser = express.Router()
[middlewareA, middlewareB, middlewareC, middlewareD, usersController]
  .forEach(handleUser.use)

app.get('/users', handleUser)
```

**Comment by derekstavis** [comment]
I continue believing that splitting arguments in parenthesis is a good pattern.

I will be closing this issue since we've merged #19.
